src/units/abilities.cpp

```diff
@@ -1784,11 +1784,9 @@ effect::effect(const unit_ability_list& list, int def, bool backstab, const_atta
 		const std::string& effect_id = cfg[cfg["id"].empty() ? "name" : "id"];
 
 		if (!cfg["backstab"].blank()) {
-			deprecated_message("backstab= in weapon specials", DEP_LEVEL::FOR_REMOVAL, {1, 19, 0}, "Use [filter_opponent] with a formula instead; the code can be found in data/core/macros/abilities.cfg in the WEAPON_SPECIAL_BACKSTAB macro.");
+			deprecated_message("backstab= in weapon specials", DEP_LEVEL::INDEFINITE, "", "Use [filter_opponent] with a formula instead; the code can be found in data/core/macros/abilities.cfg in the WEAPON_SPECIAL_BACKSTAB macro.");
```

Comment:
This line is only reached when the ability is active, if it's inactive then it's already been filtered out of the list before this function is called.

That's neither a review-failure nor a regression from the previous code, but it was a quirk when testing.

Reply:
for me, the message must be showed when player whant fight with unit owner of backstab, and not each time what unit select. thiswhy i not moved message.

Reply:
The target audience should be UMC authors, rather than players. When WML needs to be updated, only the UMC author can do that update, which is why they're the target audience. Telling the players that it doesn't work is helpful to explain why things don't work as they expect, but the target should be to alert UMC authors when they're porting to new versions of Wesnoth. Ideally it would be shown to a UMC author who's doing a sanity-check of their era or their campaign, even if they're using debug to skip a lot of the combat.

The `deprecated_message` framework should already ensure that a player only sees it once per game.

Reply:
ok, i will moves message for what it was showed immediatly